### PR TITLE
Shaokai/sa transfer learning

### PR DESCRIPTION
**Pull request by yeshaokai**
SuperAnimal transfer learning API, example and docs

### CODE REVIEW

File: examples/openfield-Pranav-2018-10-30/config.yaml [hunk 1]
```diff
@@ -1,12 +1,14 @@
-# Project definitions (do not edit)
+    # Project definitions (do not edit)
```
Comment (jeylau) [inline examples/openfield-Pranav-2018-10-30/config.yaml]: I'd revert changes automatically made to this file.

File: examples/testscript_superanimal_transfer_learning.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+"""
+import deeplabcut
+import os
+
+print (deeplabcut.__file__)
+if __name__ == "__main__":
+
+
+    superanimal_name = "superanimal_topviewmouse"
+    config_path = os.path.join(os.getcwd(), "examples", "openfield-Pranav-2018-10-30", "config.yaml")
```
Comment (jeylau) [inline examples/testscript_superanimal_transfer_learning.py]:
```suggestion
    basepath = os.path.dirname(os.path.realpath(__file__))
    config_path = os.path.join(basepath, "openfield-Pranav-2018-10-30", "config.yaml")
```

**Review by jeylau — CHANGES_REQUESTED** [review]
Two small changes to bring and looking good!

**Comment by yeshaokai** [comment]
Changes made @jeylau

**Review by jeylau — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]

**Comment by MMathisLab** [comment]
This looks great to me, with clear docs: all clear @n-poulsen / @AlexEMG ?

File: deeplabcut/generate_training_dataset/multiple_individuals_trainingsetmanipulation.py [hunk 3]
```diff
@@ -95,6 +95,7 @@ def format_multianimal_training_data(
     return train_data
 
 
+
```
Comment (n-poulsen) [inline deeplabcut/generate_training_dataset/multiple_individuals_trainingsetmanipulation.py]: there should only be [2 blank lines](https://peps.python.org/pep-0008/#blank-lines) between functions :)

```suggestion
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 4]
```diff
@@ -717,6 +719,7 @@ def to_matlab_cell(array):
     return train_data, matlab_data
 
 
+
```
Comment (n-poulsen) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: [2 blank lines](https://peps.python.org/pep-0008/#blank-lines) again


```suggestion
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 5]
```diff
@@ -728,6 +731,7 @@ def create_training_dataset(
     net_type=None,
     augmenter_type=None,
     posecfg_template=None,
+    superanimal_name = ""
```
Comment (n-poulsen) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: just a bit of styling to follow the rest of the file

```suggestion
    superanimal_name="",
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 6]
```diff
@@ -832,16 +840,28 @@ def create_training_dataset(
 
     # Loading metadata from config file:
     cfg = auxiliaryfunctions.read_config(config)
+    dlc_root_path = auxiliaryfunctions.get_deeplabcut_path()
+    
+    if superanimal_name!="":
```
Comment (n-poulsen) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: I think all these suggestions can be applied at once by running black on the code.

```suggestion
    if superanimal_name != "":
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 7]
```diff
@@ -832,16 +840,28 @@ def create_training_dataset(
...
+                dlc_root_path,
+                "pose_estimation_tensorflow",
+                "superanimal_configs",
+                supermodels[superanimal_name],
+            )
+
+
     if posecfg_template:
-        if not posecfg_template.endswith("pose_cfg.yaml"):
+        if not posecfg_template.endswith("pose_cfg.yaml") and not posecfg_template.endswith("superquadruped.yaml") and not posecfg_template.endswith("supertopview.yaml"):
```
Comment (n-poulsen) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: this line is too long, apply `black`

File: deeplabcut/modelzoo/api/spatiotemporal_adapt.py [hunk 8]
```diff
@@ -119,7 +119,10 @@ def before_adapt_inference(self, make_video=False, **kwargs):
                 customized_test_config=self.customized_pose_config,
             )
         if kwargs.pop("plot_trajectories", True):
-            _plot_trajectories(datafiles[0])
+            if len(datafiles) == 0:
+                print ('No data files found for plotting trajectory')
```
Comment (n-poulsen) [inline deeplabcut/modelzoo/api/spatiotemporal_adapt.py]:
```suggestion
                print('No data files found for plotting trajectory')
```

File: deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py [hunk 9]
```diff
@@ -45,7 +45,8 @@ def train(
     traintime_resize=False,
     video_path="",
     superanimal=None,
-    trim_ends = None  # trim the both ends of the video for video adaptation
+    trim_ends = None,  # trim the both ends of the video for video adaptation
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_tensorflow/core/train_multianimal.py]: `trim_ends` is not used in the function

**Review by n-poulsen — CHANGES_REQUESTED** [review]
`black` needs to be run on the code, as there are a few style changes to update (i suggested some changes but there are more).

I'd also like us to change the commit history for the PR before merging. There are heavy artifacts that we added and then removed from the branch (removed in commits [removed artifacts](https://github.com/DeepLabCut/DeepLabCut/pull/2389/commits/3c745ea2ca5f159d99922344bfb8a767041409f9) and [removed more artifacts](https://github.com/DeepLabCut/DeepLabCut/pull/2389/commits/78b2591dfbcc6a84bdc5af21bed4d56d5700bc3a)), but they're still in the git history.

**Comment by yeshaokai** [comment]
@n-poulsen  ran black and removed unused parameters

**Comment by MMathisLab** [comment]
Let's fix merge conflicts?

**Comment by jeylau** [comment]
Thanks @yeshaokai for running black! I've fixed the merge conflict; looking good to me now.

**Pull request merged**